distage/distage-core/src/test/scala/izumi/distage/dsl/DSLTest.scala

```diff
@@ -318,6 +321,82 @@ class DSLTest extends WordSpec {
       assert(definition1.bindings.map(_.tags.strings) == Set(Set("tag1"), Set("tag2")))
       assert(definition2.bindings.map(_.tags.strings) == Set(Set("tag1", "tag2")))
     }
+
+    "support binding to multiple interfaces" in {
+      import BasicCase6._
+
+      val implXYZ = new ImplXYZ
+
+      val definition = new ModuleDef {
+        bind[ImplXYZ]
+          .to[TraitX]
+          .to[TraitY]
+          .to[TraitZ]
+      }
+
+      assert(definition === Module.make(
+        Set(
+          Bindings.binding[ImplXYZ]
+          , Bindings.reference[TraitX, ImplXYZ]
+          , Bindings.reference[TraitY, ImplXYZ]
+          , Bindings.reference[TraitZ, ImplXYZ]
+        )
+      ))
+
+      val definitionEffect = new ModuleDef {
+        bindEffect[Identity, ImplXYZ](implXYZ).to[TraitX].to[TraitY].to[TraitZ]
+      }
+
+      assert(definitionEffect === Module.make(
+        Set(
+          SingletonBinding(DIKey.get[ImplXYZ], ImplDef.EffectImpl(SafeType.get[ImplXYZ], SafeType.getK[Identity],
+            ImplDef.InstanceImpl(SafeType.get[ImplXYZ], implXYZ)))
+          , Bindings.reference[TraitX, ImplXYZ]
+          , Bindings.reference[TraitY, ImplXYZ]
+          , Bindings.reference[TraitZ, ImplXYZ]
+        )
+      ))
+
+      val definitionResource = new ModuleDef {
+        bindResource[DIResource.Simple[ImplXYZ], ImplXYZ].to[TraitX].to[TraitY].to[TraitZ]
+      }
+
+      assert(definitionResource === Module.make(
+        Set(
+          SingletonBinding(DIKey.get[ImplXYZ], ImplDef.ResourceImpl(SafeType.get[ImplXYZ], SafeType.getK[Identity],
+            ImplDef.TypeImpl(SafeType.get[DIResource.Simple[ImplXYZ]])))
+          , Bindings.reference[TraitX, ImplXYZ]
+          , Bindings.reference[TraitY, ImplXYZ]
+          , Bindings.reference[TraitZ, ImplXYZ]
+        )
+      ))
+
+    }
+
+    "support bindings to multiple interfaces (injector test)" in {
+      import BasicCase6._
+
+      val definition = PlannerInput.noGc(new ModuleDef {
+        bind[ImplXYZ].named("my-impl")
+          .to[TraitX]
+          .to[TraitY]("Y")
+      })
+
+      val defWithoutSugar = PlannerInput.noGc(new ModuleDef {
+        make[ImplXYZ].named("my-impl")
+        make[TraitX].using[ImplXYZ]("my-impl")
+        make[TraitY].named("Y").using[ImplXYZ]("my-impl")
+      })
+
+      val injector = mkInjector()
+      val plan1 = injector.plan(definition)
+      val plan2 = injector.plan(defWithoutSugar)
+      assert(plan1.definition == plan2.definition)
+
+      val context = injector.produceUnsafe(plan1)
+      assert(context.get[TraitX].isInstanceOf[ImplXYZ])
```

Comment:
Could you use `eq` here to make sure it's the same exact instance?

Reply:
Done.